#! /bin/sh -e
## 04_qsql_odbc.dpatch by Martin Loschwitz <madkiss@madkiss.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make the package build with libiodbc2-dev instead of unixodbc-dev

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

diff -ruN qt-x11-free-3.2.1.old/plugins/src/sqldrivers/odbc/odbc.pro qt-x11-free-3.2.1/plugins/src/sqldrivers/odbc/odbc.pro
--- qt-x11-free-3.2.1.old/plugins/src/sqldrivers/odbc/odbc.pro	2003-06-19 22:07:43.000000000 +0200
+++ qt-x11-free-3.2.1/plugins/src/sqldrivers/odbc/odbc.pro	2003-08-29 16:15:21.000000000 +0200
@@ -15,7 +15,7 @@
 unix {
 	OBJECTS_DIR	= .obj
 	!contains( LIBS, .*odbc.* ) {
-	    LIBS 	*= -lodbc
+	    LIBS 	*= -liodbc
 	}
 }
 
diff -ruN qt-x11-free-3.2.1.old/src/sql/drivers/odbc/debian_qsql_odbc.h qt-x11-free-3.2.1/src/sql/drivers/odbc/debian_qsql_odbc.h
--- qt-x11-free-3.2.1.old/src/sql/drivers/odbc/debian_qsql_odbc.h	1970-01-01 01:00:00.000000000 +0100
+++ qt-x11-free-3.2.1/src/sql/drivers/odbc/debian_qsql_odbc.h	2003-08-29 16:15:21.000000000 +0200
@@ -0,0 +1,10 @@
+#ifdef UNICODE
+typedef SQLWCHAR        SQLTCHAR;
+#else
+typedef SQLCHAR         SQLTCHAR;
+#endif
+
+#define SQL_WCHAR              (-8)
+#define SQL_WVARCHAR           (-9)
+#define SQL_WLONGVARCHAR       (-10)
+#define SQL_C_WCHAR            SQL_WCHAR
diff -ruN qt-x11-free-3.2.1.old/src/sql/drivers/odbc/qsql_odbc.h qt-x11-free-3.2.1/src/sql/drivers/odbc/qsql_odbc.h
--- qt-x11-free-3.2.1.old/src/sql/drivers/odbc/qsql_odbc.h	2003-06-19 22:00:21.000000000 +0200
+++ qt-x11-free-3.2.1/src/sql/drivers/odbc/qsql_odbc.h	2003-08-29 16:16:42.000000000 +0200
@@ -74,11 +74,8 @@
 #  undef _MSC_VER
 #endif
 
-#ifndef Q_ODBC_VERSION_2
-#include <sqlucode.h>
-#endif
-
 #include <sqlext.h>
+#include "debian_qsql_odbc.h"
 
 class QODBCPrivate;
 class QODBCDriver;
diff -ruN qt-x11-free-3.2.1.old/src/sql/qt_sql.pri qt-x11-free-3.2.1/src/sql/qt_sql.pri
--- qt-x11-free-3.2.1.old/src/sql/qt_sql.pri	2003-07-08 11:33:23.000000000 +0200
+++ qt-x11-free-3.2.1/src/sql/qt_sql.pri	2003-08-29 16:15:21.000000000 +0200
@@ -115,7 +115,7 @@
 
 		unix {
 			!contains( LIBS, .*odbc.* ) {
-				LIBS        *= -lodbc
+				LIBS        *= -liodbc
 			}
 		}
 
diff -ruN qt-x11-free-3.3.3.old/src/sql/drivers/odbc/qsql_odbc.cpp qt-x11-free-3.3.3/src/sql/drivers/odbc/qsql_odbc.cpp
--- qt-x11-free-3.3.3.old/src/sql/drivers/odbc/qsql_odbc.cpp	2004-05-27 15:43:13.000000000 +0000
+++ qt-x11-free-3.3.3/src/sql/drivers/odbc/qsql_odbc.cpp	2004-10-31 09:35:06.000000000 +0000
@@ -57,7 +57,7 @@
 # define QSQLLEN SQLINTEGER
 # define QSQLULEN SQLUINTEGER
 #else
-# ifndef Q_OS_WIN64
+# if 0
 #  define SQLLEN SQLINTEGER
 #  define SQLULEN SQLUINTEGER
 # endif
